osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSliderTail.cs

```diff
@@ -125,8 +126,15 @@ protected override void UpdateHitStateTransforms(ArmedState state)
 
         protected override void CheckForResult(bool userTriggered, double timeOffset)
         {
-            if (!userTriggered && timeOffset >= 0)
-                ApplyResult(r => r.Type = Tracking ? r.Judgement.MaxResult : r.Judgement.MinResult);
+            if (userTriggered)
+                return;
+
+            // The player needs to have engaged in tracking at any point after the tail leniency cutoff.
+            // An actual tick miss should only occur if reaching the tick itself.
+            if (timeOffset >= SliderEventGenerator.TAIL_LENIENCY && Tracking)
+                ApplyResult(r => r.Type = r.Judgement.MaxResult);
+            else if (timeOffset >= 0)
```

Comment:
Would probably be more comfortable with

```suggestion
            else if (timeOffset > 0)
```

Not sure it's gonna matter in practice; it might on legacy replays where all time instants are non-fractional.